File: minecraft-web-manager/backend/src/scripts/create-admin.js
import readline from 'readline';
import dotenv from 'dotenv';
import { runMigrations } from '../database/db.js';
import User from '../models/User.js';

// Cargar variables de entorno
dotenv.config();

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function question(query) {
  return new Promise(resolve => rl.question(query, resolve));
}

async function createAdminUser() {
  console.log('\n🔐 Crear Usuario Administrador');
  console.log('='.repeat(50));

  try {
    // Ejecutar migraciones
    console.log('\n📦 Verificando base de datos...');
    runMigrations();

    // Verificar si ya existe un usuario admin
    const existingAdmins = User.findAll().filter(u => u.role === 'admin');
    if (existingAdmins.length > 0) {
      console.log('\n⚠️  Ya existen usuarios administradores:');
      existingAdmins.forEach(admin => {
        console.log(`   - ${admin.username} (ID: ${admin.id})`);
      });

      const confirm = await question('\n¿Deseas crear otro usuario admin? (s/n): ');
      if (confirm.toLowerCase() !== 's' && confirm.toLowerCase() !== 'si') {
        console.log('\n❌ Operación cancelada');
        rl.close();
        return;
      }
    }

    // Solicitar datos del usuario
    console.log('\nIngresa los datos del nuevo usuario administrador:\n');

    const username = await question('Username: ');
    if (!username || username.trim().length === 0) {
      console.log('\n❌ El username no puede estar vacío');
      rl.close();
      return;
    }

    // Verificar si el username ya existe
    const existingUser = User.findByUsername(username);
    if (existingUser) {
      console.log(`\n❌ El usuario "${username}" ya existe`);
      rl.close();
      return;
    }

    const password = await question('Password (mínimo 6 caracteres): ');
    if (!password || password.length < 6) {
      console.log('\n❌ La contraseña debe tener al menos 6 caracteres');
      rl.close();
      return;
    }

    const confirmPassword = await question('Confirmar password: ');
    if (password !== confirmPassword) {
      console.log('\n❌ Las contraseñas no coinciden');
      rl.close();
      return;
    }

    // Crear usuario
    console.log('\n📝 Creando usuario...');
    const user = User.create({
      username,
      password,
      role: 'admin'
    });

    console.log('\n✅ Usuario administrador creado exitosamente!');
    console.log('\nDetalles del usuario:');
    console.log(`   - ID: ${user.id}`);
    console.log(`   - Username: ${user.username}`);
    console.log(`   - Role: ${user.role}`);
    console.log(`   - Creado: ${user.created_at}`);
    console.log('\n✨ Ya puedes iniciar sesión con este usuario');

  } catch (error) {
    console.error('\n❌ Error al crear usuario:', error);
  } finally {
    rl.close();
  }
}

// Ejecutar
createAdminUser();

File: minecraft-web-manager/backend-python/app/scripts/create_admin.py
"""
Script para crear usuario administrador
"""
import sys
import os
from getpass import getpass

# Agregar el directorio raíz al path
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.dirname(__file__))))

from app.db.session import SessionLocal, init_db
from app.models.user import User
from app.core.security import get_password_hash


def create_admin():
    """Crear usuario administrador interactivamente"""
    
    print("\n" + "="*50)
    print("  Crear Usuario Administrador")
    print("="*50 + "\n")
    
    # Inicializar DB primero
    init_db()
    
    # Solicitar datos
    username = input("Username: ").strip()
    
    if not username:
        print("❌ El username no puede estar vacío")
        sys.exit(1)
    
    # Verificar si ya existe
    db = SessionLocal()
    try:
        existing = db.query(User).filter(User.username == username).first()
        if existing:
            print(f"❌ El usuario '{username}' ya existe")
            sys.exit(1)
        
        # Solicitar password
        while True:
            password = getpass("Password: ")
            password_confirm = getpass("Confirmar password: ")
            
            if not password:
                print("❌ El password no puede estar vacío")
                continue
            
            if password != password_confirm:
                print("❌ Los passwords no coinciden, intenta de nuevo")
                continue
            
            if len(password) < 6:
                print("❌ El password debe tener al menos 6 caracteres")
                continue
            
            # Bcrypt tiene límite de 72 bytes
            if len(password.encode('utf-8')) > 72:
                print("⚠️  Password muy largo, se truncará a 72 bytes")
                password = password.encode('utf-8')[:72].decode('utf-8', errors='ignore')
            
            break
        
        # Crear usuario
        user = User(
            username=username,
            password_hash=get_password_hash(password),
            role="admin"
        )
        
        db.add(user)
        db.commit()
        db.refresh(user)
        
        print(f"\n✅ Usuario admin '{username}' creado exitosamente!")
        print(f"   ID: {user.id}")
        print(f"   Role: {user.role}")
        print(f"   Creado: {user.created_at}\n")
        
    except Exception as e:
        print(f"\n❌ Error al crear usuario: {e}\n")
        db.rollback()
        sys.exit(1)
    finally:
        db.close()


if __name__ == "__main__":
    create_admin()
